File: solutions/zig/armstrong-numbers/2/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    var sum: u128 = 0;

    var digits: u32 = 0;
    var temp: u128 = num;

    while (temp != 0) {
        digits += 1;
        temp /= 10;
    }

    temp = num;
    while (temp != 0) {
        sum += std.math.pow(u128, temp % 10, digits);
        temp /= 10;
    }

    return sum == num;
}

File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrongnumbers

import "math"

func IsNumber(n int) bool {
	var sum float64
	power := int(math.Log10(float64(n))) + 1

	for temp := n; temp != 0; temp /= 10 {
		sum += math.Pow(float64(temp%10), float64(power))
	}
	return int(sum) == n

}
